### PR TITLE
Ruby 2.7: Kernel#lambda with no block in a method called with a block raises an exception

### PR DESCRIPTION
**Pull request by ssnickolay**
Didn't find an MRI test for this 🤔

### CODE REVIEW

File: src/main/java/org/truffleruby/core/kernel/KernelNodes.java [hunk 1]
```diff
@@ -1193,31 +1193,9 @@ protected boolean isATypeError(Object self, Object module) {
 
         @Child private WarnNode warnNode;
```
Comment (eregon) [inline src/main/java/org/truffleruby/core/kernel/KernelNodes.java:1194]: This and `warnProcWithoutBlock()` below can be removed too.

**Comment by eregon** [comment]
> Didn't find an MRI test for this

That's fine. When changing MRI, it's OK to add only a ruby/spec and not an MRI test, maybe that's what happened here.

File: src/main/java/org/truffleruby/core/kernel/KernelNodes.java (hunk 1 above)
Comment (ssnickolay) [reply src/main/java/org/truffleruby/core/kernel/KernelNodes.java:1193]: I forgot to push `$ jt checkstyle` result 🙈 Fixed

Comment (eregon) [reply src/main/java/org/truffleruby/core/kernel/KernelNodes.java:1193]: I still see the field above in the diff, could you remove it?

Comment (ssnickolay) [reply src/main/java/org/truffleruby/core/kernel/KernelNodes.java]: Meh, this one I forgot to remove; Sorry, I should be a bit more attentive. Done

**Review by eregon — APPROVED** [review]
Thanks :)

**Pull request merged**